### PR TITLE
Add an implicit zero after trailing semicolon

### LINKED ISSUE #23: Interpret a trailing semicolon as an implicit '0' parameter
From what I can find, escape sequence parameters are "optional numeric values separated by semicolons".

As the values are separated by semicolons, it makes sense that if a sequence ends with a semicolon, the terminal should expect another parameter. Because the parameter is an optional number, it should default to 0.

For example, the code `\e[4;m` should have the same effect as `\e[4;0m`, that is, none at all.

I tested this specific sequence in kitty, termite, xterm, st and alacritty, and only alacritty draws the following text underlined.

It should be a simple fix and I will include it in my PR for #22

### PR DESCRIPTION
**Pull request by VojtechStep**
fixes #23

### CODE REVIEW

**Comment by VojtechStep** [comment]
Seems alright and checks pass, let me change it real quick.

File: src/lib.rs [hunk 1]
```diff
@@ -294,11 +292,10 @@ impl Parser {
             },
             Action::Unhook => performer.unhook(),
             Action::CsiDispatch => {
-                if self.collecting_param {
-                    let idx = self.num_params;
-                    self.params[idx] = self.param;
-                    self.num_params += 1;
-                }
+
```
Comment (chrisduerr) [inline src/lib.rs]: There's an excess newline here.

File: src/lib.rs [hunk 2]
```diff
@@ -557,6 +551,19 @@ mod tests {
 
     }
 
+    #[test]
+    fn parse_trailing_semi() {
```
Comment (chrisduerr) [inline src/lib.rs]: I know that you've probably used the other tests to come up with this name, but I think it's a bit unclear.

I feel like `csi_parse_trailing_semicolon` might be a bit clearer here.

Comment (VojtechStep) [reply src/lib.rs]: Sure, I was just trying to pick up on the convention :)

Comment (chrisduerr) [reply src/lib.rs]: > Sure, I was just trying to pick up on the convention :)

Yeah, which I think was a good idea. But I feel like the conventions used here aren't optimal here.

**Comment by jwilm** [comment]
Sweet PR! Gotta love that some branching and state were removed. :smile:

**Pull request merged**